### ISSUE TITLE
win_acl: "domain\user is not a valid user or group on the host machine or domain" when user is added to server

### ISSUE THREAD

**Issue by andmos** [ansible/ansible#29005]
##### ISSUE TYPE
 - Bug Report
 
##### COMPONENT NAME
`win_acl`

##### ANSIBLE VERSION

```
ansible 2.3.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### CONFIGURATION
All default

##### OS / ENVIRONMENT
Running from Ubuntu Trusty 14.04
Provisioning Windows Server 2012R2 

##### SUMMARY
When trying to give domain user explicit permissions to a folder: 
```
- name: Give domainUser access to C:\\oracle
  win_acl:
    path: C:\\oracle
    user: domain-ad\domainUser
    rights: FullControl
    type: allow
    state: present
  tags: permission
```

The play errors with 
```
fatal: [vd-cdlab11.domain-ad.local]: FAILED! => {
    "changed": false,
    "failed": true,
    "msg": "domain-ad\\domainUser is not a valid user or group on the host machine or domain"
}
```
The machine is member of the domain (here censored) and the user exists, since I use it when RDP'ing in to the server. I also have a role that gives the specific user login as a service permission and adds it to the local Administrators-group. Checking the server, I see the user in this group so I know it is there. 

My `group_vars` file show how Ansible connects: 

```
ansible_user: ansibleadmin
ansible_password: *********
ansible_port: 5986
ansible_connection: winrm
ansible_winrm_server_cert_validation: ignore
```
We connect via a local user on the machine, not via Kerberos. Should not have anything to say. 

##### STEPS TO REPRODUCE

Create a folder on the Windows machine, like `C:\oracle`

```yaml
- name: Give domainUser access to C:\\oracle
  win_acl:
    path: C:\\oracle
    user: domain-ad\domainUser
    rights: FullControl
    type: allow
    state: present
  tags: permission

```

##### EXPECTED RESULTS
The domain user should get permission to the folder. 

##### ACTUAL RESULTS
```
TASK [Gathering Facts] *****************************************************************************************************************************************
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/windows/setup.ps1
<vd-cdlab11.domain-ad.local> ESTABLISH WINRM CONNECTION FOR USER: ansibleadmin on PORT 5986 TO vd-cdlab11.domain-ad.local
EXEC (via pipeline wrapper)
ok: [vd-cdlab11.domain-ad.local]
META: ran handlers

TASK [Oracle12c : Give "spud" access to C:\\oracle] ************************************************************************************************************
task path: /vagrant/roles/Oracle12c/tasks/main.yaml:25
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/windows/win_acl.ps1
<vd-cdlab11.domain-ad.local> ESTABLISH WINRM CONNECTION FOR USER: ansibleadmin on PORT 5986 TO vd-cdlab11.domain-ad.local
EXEC (via pipeline wrapper)
fatal: [vd-cdlab11.domain-ad.local]: FAILED! => {
    "changed": false,
    "failed": true,
    "msg": "domain-ad\\domainUser is not a valid user or group on the host machine or domain"
}
	to retry, use: --limit @/vagrant/OctopusDeploy.retry
```

**Comment by ansibot** [comment]
cc @schwartzmx @trondhindenes
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @schwartzmx @trondhindenes
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jhawkesworth** [comment]
@andmos - please could you test https://github.com/ansible/ansible/pull/28979 as I think it might fix your problem?

**Comment by andmos** [comment]
Looks to me like #28979 might do the trick. I will try it out when I find the time!

**Comment by jborean93** [comment]
I've just tested this myself with a domain account. I can confirm the current implementation of getting a domain SID through `[ADSISearcher]` fails when using a local account or when if credential delegation isn't support when using a domain account.

I can confirm the changes in that PR fixes it up as it no longer using `[ADSISearcher]` and even works on local accounts searching for a domain account.

I am going to close the issue as this has been merged and will be available in the next 2.4 release.
